src/NotesLoader.cpp

```diff
@@ -11,63 +11,87 @@
 #include <cstddef>
 #include <vector>
 
-
 void NotesLoader::GetMainAndSubTitlesFromFullTitle( const RString &sFullTitle, RString &sMainTitleOut, RString &sSubTitleOut )
 {
-	const RString sLeftSeps[]  = { "\t", " -", " ~", " (", " [" };
+	static const std::string_view sLeftSeps[] = { "\t", " -", " ~", " (", " [" };
+	size_t fullTitleSize = sFullTitle.size();
 
-	for( unsigned i=0; i<ARRAYLEN(sLeftSeps); i++ )
+	for (const auto& sep : sLeftSeps)
 	{
-		size_t iBeginIndex = sFullTitle.find( sLeftSeps[i] );
-		if( iBeginIndex == std::string::npos )
-			continue;
-		sMainTitleOut = sFullTitle.Left( (int) iBeginIndex );
-		sSubTitleOut = sFullTitle.substr( iBeginIndex+1, sFullTitle.size()-iBeginIndex+1 );
-		return;
+		size_t iBeginIndex = sFullTitle.find(sep);
+		if (iBeginIndex != std::string::npos)
+		{
+			sMainTitleOut = sFullTitle.Left(static_cast<int>(iBeginIndex));
+			sSubTitleOut = sFullTitle.substr(iBeginIndex + sep.size(), fullTitleSize - iBeginIndex - sep.size());
+			return;
+		}
 	}
 	sMainTitleOut = sFullTitle;
 	sSubTitleOut = "";
-};
+}
 
-bool NotesLoader::LoadFromDir( const RString &sPath, Song &out, std::set<RString> &BlacklistedImages, bool load_autosave )
+// On first run, let's assume we have all of DWI, BMS, and KSF.
+// In reality, most users don't have these file types at all.
+// So, if they're not detected, we'll set the flag to false,
+// and then the checks for those file types won't run again.
+static bool dwiDetected = true;
+static bool bmsDetected = true;
+static bool ksfDetected = true;
```

Comment:
undecided if these should be non-static booleans within the scope of the function...?  As far as I can tell, that doesn't change anything.  all my DWI's load and play as expected either way.